File: UIViewControllerTransitions/Classes/UIMaskedImageView.m
//
//  UIMaskedImageView.m
//  ModalTransitionAnimator
//
//  Created by Steve Kim on 5/12/16.
//

#import "UIMaskedImageView.h"

@implementation UIMaskedImageView

- (void)layoutSubviews {
    [super layoutSubviews];
    
    CAAnimation* animation = [self.layer animationForKey:@"bounds"];
    if (animation) {
        [CATransaction begin];
        [CATransaction setAnimationDuration:animation.duration];
    }
    
    self.layer.mask.bounds = self.layer.bounds;
    if (animation) {
        [CATransaction commit];
    }
}

@end

File: AnimatedTransitionKit/Classes/Views/UIMaskedImageView.swift
//
//  UIMaskedImageView.swift
//  Pods
//
//  Created by Minwoo on 9/3/25.
//

import UIKit

class UIMaskedImageView: UIImageView {
    override func layoutSubviews() {
        super.layoutSubviews()

        if let animation = layer.animation(forKey: "bounds") {
            CATransaction.begin()
            CATransaction.setAnimationDuration(animation.duration)
        }

        layer.mask?.bounds = layer.bounds

        if let _ = layer.animation(forKey: "bounds") {
            CATransaction.commit()
        }
    }
}
